lib/mix/tasks/oban_web.install.ex

```diff
@@ -0,0 +1,167 @@
+defmodule Mix.Tasks.ObanWeb.Install.Docs do
+  @moduledoc false
+
+  def short_doc do
+    "Installs ObanWeb dashboard into your Phoenix application"
+  end
+
+  def example do
+    "mix oban_web.install"
+  end
+
+  def long_doc do
+    """
+    #{short_doc()}
+
+    This task configures your Phoenix application to use the ObanWeb dashboard:
+
+    * Adds the required `Oban.Web.Router` import
+    * Sets up the dashboard route at "/oban" within the :dev_routes conditional
+
+    ## Example
+
+    ```bash
+    #{example()}
+    """
+  end
+end
+
+if Code.ensure_loaded?(Igniter) do
+  defmodule Mix.Tasks.ObanWeb.Install do
+    @shortdoc "#{__MODULE__.Docs.short_doc()}"
+
+    @moduledoc __MODULE__.Docs.long_doc()
+    use Igniter.Mix.Task
+
+    @impl Igniter.Mix.Task
+    def info(_argv, _composing_task) do
+      %Igniter.Mix.Task.Info{
+        group: :oban,
+        installs: [{:oban, "~> 2.0"}],
+        example: __MODULE__.Docs.example()
+      }
+    end
+
+    @impl Igniter.Mix.Task
+    def igniter(igniter) do
+      case Igniter.Libs.Phoenix.select_router(igniter) do
+        {igniter, nil} ->
+          igniter
+          |> Igniter.add_warning("""
+          No Phoenix router found, Phoenix Liveview is needed for ObanWeb
+          """)
+
+        {igniter, router} ->
+          update_router(igniter, router)
+      end
+    end
+
+    defp update_router(igniter, router) do
+      case Igniter.Project.Module.find_and_update_module(
+             igniter,
+             router,
+             &do_update_router(igniter, &1)
+           ) do
+        {:ok, igniter} ->
+          igniter
+
+        {:error, igniter} ->
+          igniter
+          |> Igniter.add_warning("""
+          Something went wrong, please check the ObanWeb install docs for manual setup instructions
+          """)
+      end
```

Comment:
Formatting nitpicks only:

```suggestion
      zipper = &do_update_router(igniter, &1)

      case Igniter.Project.Module.find_and_update_module(igniter, router, zipper) do
        {:ok, igniter} ->
          igniter

        {:error, igniter} ->
          Igniter.add_warning(igniter, """
          Something went wrong, please check the Oban Web install docs for manual setup instructions
          """)
      end
```